File: python/gamex/platforms/eginx/eginx_render.py
from __future__ import annotations
import os
from openstk.core import ISource, IHaveSource
from openstk.platforms.eginx.gfx import TestTriRenderer

# typedefs
class IOpenGfx: pass
class Renderer: pass

# EginXRenderer
class EginXRenderer:
    @staticmethod
    def createRenderer(parent: object, gfx: list[IOpenGfx], source: ISource, obj: object, type: str) -> Renderer:
        surf = parent.surface
        if isinstance(obj, IHaveSource): source = obj.source
        match type:
            case 'TestTri': return TestTriRenderer(gfx, source, obj, surf)
            case _: return None

File: dotnet/Platforms/GameX.Platform.EginX/EginX_Render.cs
using OpenStack;
using OpenStack.Gfx;
using OpenStack.Gfx.EginX;
#pragma warning disable CS9113

namespace GameX.Platforms.EginX;

public static class EginXRenderer {
    public static Renderer CreateRenderer(object parent, IOpenGfx[] gfx, ISource source, object obj, string type) {
        if (obj is IHaveSource z) source = z.Source;
        return type switch {
            "TestTri" => new TestTriRenderer(gfx, source, obj),
            "Texture" => new SpriteRenderer(gfx, source, obj),
            //"Object" => new ObjectRenderer(gfx, source, obj),
            _ => default
        };
    }
}
